### PR TITLE
Fixes #3443, Actionbar for edit waypoint

### LINKED ISSUE #3443: Waypoint improvements
Working with waypoints is more awkward than it should be. I propose the following changes to make the experience 'smoother':
- Remove the save button from edit waypoint, add save and cancel to 'action bar', make 'save' the default action on 'back'
- Swap short and long press in waypoint list. Short-press - edit, long-press - menu

### PR DESCRIPTION
**Pull request by rsudev**
- remove save button, add action bar items
- add asynchronous notification for cache details when the save is done

### CODE REVIEW

**Comment by Bananeweizen** [comment]
Just a short question: Do the 2 buttons appear at the right hand side of the action bar? I'm asking because there are applications which manage to show Okay/Cancel buttons over the complete width of the actionbar. I'm not sure of the implementation details of that however.

**Comment by rsudev** [comment]
No, they are just normal 'action items' in the top right, 'Save' being the the rightmost and cancel to the left of that.

**Comment by rsudev** [comment]
Here's a screenshot:
screen

**Comment by rsudev** [comment]
Ups, please do not merge - there is a glaring oversight in the save triggering...

**Comment by Lineflyer** [comment]
UI looks very good IMO.

**Comment by rsudev** [comment]
Ok, fixed now. Ready for review

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 1]
```diff
@@ -365,160 +399,151 @@ private WaypointType getSelectedWaypointType() {
...
+        final String latText = buttonLat.getText().toString();
+        final String lonText = buttonLon.getText().toString();
+
+        if (StringUtils.isBlank(bearingText) && StringUtils.isBlank(distanceText)
+                && StringUtils.isBlank(latText) && StringUtils.isBlank(lonText)) {
+            Dialogs.message(EditWaypointActivity.this, R.string.err_point_no_position_given_title, R.string.err_point_no_position_given);
+            return;
+        }
+
+        Geopoint coords;
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/EditWaypointActivity.java]: Can't this one be `final`?

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 2]
```diff
@@ -365,160 +399,151 @@ private WaypointType getSelectedWaypointType() {
...
                 return;
             }
+        } else {
+            coords = Sensors.getInstance().currentGeo().getCoords();
+        }
 
-            Geopoint coords;
+        if (StringUtils.isNotBlank(bearingText) && StringUtils.isNotBlank(distanceText)) {
+            // bearing & distance
+            double bearing;
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/EditWaypointActivity.java]: Can probably be `final`.

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 3]
```diff
@@ -365,160 +399,151 @@ private WaypointType getSelectedWaypointType() {
...
-            if (StringUtils.isNotBlank(latText) && StringUtils.isNotBlank(lonText)) {
-                try {
-                    coords = new Geopoint(latText, lonText);
-                } catch (final Geopoint.ParseException e) {
-                    showToast(res.getString(e.resource));
-                    return;
-                }
-            } else {
-                coords = Sensors.getInstance().currentGeo().getCoords();
+            double distance;
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/EditWaypointActivity.java]: Can probably be `final`.

**Comment by samueltardieu** [comment]
Looks fine to me. I will merge it locally and look at my `final` remarks then push it.

**Comment by rsudev** [comment]
I must admit I didn't look too detailed into the preexisting code (there I only changed the indentation). Normally Eclipse is not bad at adding 'final' where ever it may apply, but didn't catch these occurrences apparently.

**Pull request merged**